src/JustEat.StatsD/PooledUdpTransport.cs

```diff
@@ -10,13 +11,71 @@ namespace JustEat.StatsD
     /// </summary>
     public sealed class PooledUdpTransport : IStatsDTransport, IDisposable
     {
-        private readonly SimpleObjectPool<Socket> _socketPool
-            = new SimpleObjectPool<Socket>(
-                Environment.ProcessorCount,
-                pool => UdpTransport.CreateSocket());
+        private sealed class ConnectedPool : IDisposable
```

Comment:
This private class isn't trivial, so can you make it internal and put it in its own file please?